tests/testVector.cpp

```diff
@@ -73,10 +73,12 @@
 #endif
 
 template <typename T>
-int vectorTests(T test, const char* mem_space, int rank, int numRanks, MPI_Comm comm)
+int vectorTests(const char* mem_space, int rank, int numRanks, MPI_Comm comm)
```

Comment:
May be overkill, but I think we should constrain this type:
```cpp
template <class T,
          typename std::enable_if<
            std::is_base_of<hiop::tests::VectorTests, T>::value,
            bool>::type = true>
int vectorTests(const char* mem_space, int rank, int numRanks, MPI_Comm comm)
```
so you can only call this with the correct test types. This should be malformed if you call with `hiop::tests::VectorTestsInt` for example.